### ISSUE TITLE
[Swipe to refresh] Swipe action cannot be triggered

### ISSUE THREAD

**Issue by TonnyL** [google/accompanist#1028]
**Description**

**Steps to reproduce**
```kotlin
class MainActivity : ComponentActivity() {
    @ExperimentalMaterialApi
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            MyApplicationTheme {
                // A surface container using the 'background' color from the theme
                Surface(color = MaterialTheme.colors.background) {
                    SwipeRefresh(state = rememberSwipeRefreshState(isRefreshing = false), onRefresh = {}) {
                        LazyColumn(
                            contentPadding = PaddingValues(all = 10.dp),
                            modifier = Modifier.fillMaxWidth()
                        ) {
                            item {
                                Spacer(modifier = Modifier.size(size = 0.dp))
                            }

                            items(count = 10) {
                                ListItem(text = {Text(text = it.toString())})
                            }

                            item {
                                ListItem(text = {
                                    Text(text = "footer")
                                })
                            }
                        }
                    }
                }
            }
        }
    }
}
```

I think it's relevant to `contentPadding = PaddingValues(all = 10.dp)` or `item {
                                Spacer(modifier = Modifier.size(size = 0.dp))
                            }`

**Expected behavior** 

**Additional context**
compose_version = 1.0.0
accompanist-swiperefresh = 0.24.2-alpha

**Comment by sitatec** [comment]
>I think it's relevant to contentPadding = PaddingValues(all = 10.dp) or item { Spacer(modifier = Modifier.size(size = 0.dp)) }

Yes, you are right: 
- If you remove either of them, the indicator is triggered
- If you set the value of the `Spacer`'s size to >= 1 it works
- If you set both `PaddingValues` and the `Spacer`'s size to 0 it works as well 

*But for the  `PaddingValues`, only its vertical paddings have effect on the trigger (which make sense)*

But the problem doesn't come from `SwipeRefresh`. It seems like the `LazyColumn` is consuming all the user drag offset when one of the above scenarios are produced.

To see what is happening, I wrapped the  `LazyColumn` in the code @TonnyL  provided (without the `SwipeRefresh`) in a [NestedScrollConnection](https://developer.android.com/reference/kotlin/androidx/compose/ui/input/nestedscroll/NestedScrollConnection) and added some log in the  `onPreScroll` and `onPostScroll`  callbacks :

```kotlin
  class TestNestedScrollConnection : NestedScrollConnection {
  
      override fun onPreScroll(available: Offset, source: NestedScrollSource): Offset {
          println("------ onPreScroll -- Drag Available => $available")
          return super.onPreScroll(available, source)
      }
  
      override fun onPostScroll(
          consumed: Offset,
          available: Offset,
          source: NestedScrollSource
      ): Offset {
          println("------ onPostScroll -- Drag consumed => $consumed || Drag Available => $available")
          return super.onPostScroll(consumed, available, source)
      }
  }

------------------------------------------------------------------------------------------------
 
  Box(Modifier.nestedScroll(connection = TestNestedScrollConnection())) {
      LazyColumn(
          contentPadding = PaddingValues(vertical = 10.dp),
          modifier = Modifier.fillMaxWidth()
      ) {

          item {
              Spacer(modifier = Modifier.size(size = 0.dp))
          }

          items(count = 10) {
              ListItem(text = { Text(text = it.toString()) })
          }

          item {
              ListItem(text = {
                  Text(text = "footer")
              })
          }
      }
  }
```
<br/>

### For any of the scenarios, I listed above (When the indicator is not triggered) here are the logs:
```shell
------ onPreScroll -- Drag Available => Offset(0.0, 6.9)
------ onPostScroll -- Drag consumed => Offset(0.0, 6.9) || Drag Available => Offset(0.0, 0.0)
------ onPreScroll -- Drag Available => Offset(0.0, 5.7)
------ onPostScroll -- Drag consumed => Offset(0.0, 5.7) || Drag Available => Offset(0.0, 0.0)
------ onPreScroll -- Drag Available => Offset(0.0, 4.5)
------ onPostScroll -- Drag consumed => Offset(0.0, 4.5) || Drag Available => Offset(0.0, 0.0)
------ onPreScroll -- Drag Available => Offset(0.0, 3.9)
------ onPostScroll -- Drag consumed => Offset(0.0, 3.9) || Drag Available => Offset(0.0, 0.0)
```
As you can see, the `LazyColumn` is consuming all the available drag no matter how fare you drag. It is behaving like if there was an infinite number of items at the top of the list, thus it is consuming all the drag like if you are infinitely scrolling up (but there is no scroll).
>If you move the `Spacer` at the end of the `LazyColumn` the trigger works.
<br/>

### For the scenarios where the indicator is triggered (e.g. when we remove the `contentPadding`) :

```shell
------ onPreScroll -- Drag Available => Offset(0.0, 19.2)
------ onPostScroll -- Drag consumed => Offset(0.0, 0.0) || Drag Available => Offset(0.0, 19.2)
------ onPreScroll -- Drag Available => Offset(0.0, 17.4)
------ onPostScroll -- Drag consumed => Offset(0.0, 0.0) || Drag Available => Offset(0.0, 17.4)
------ onPreScroll -- Drag Available => Offset(0.0, 16.9)
------ onPostScroll -- Drag consumed => Offset(0.0, 0.0) || Drag Available => Offset(0.0, 16.9)
------ onPreScroll -- Drag Available => Offset(0.0, 23.3)
------ onPostScroll -- Drag consumed => Offset(0.0, 0.0) || Drag Available => Offset(0.0, 23.3)
------ onPreScroll -- Drag Available => Offset(0.0, 17.3)
------ onPostScroll -- Drag consumed => Offset(0.0, 0.0) || Drag Available => Offset(0.0, 17.3)
```
Here, the `LazyColumn` isn't consuming any of the Offset, this is the normal/expected behaviour since there are not enough items in the `LazyColumn` to perform a scroll.

Basically `SwipeRefresh` need drag it can consume in order to trigger the indicator, since the `LazyColumn` is consuming all of it, the `SwipeRefresh` can't trigger the indicator because from its point of view there is no drag being performed. So this is probably an issue in the `LazyColumn`.

*compose.version = "1.2.0-alpha03"*

**Comment by TonnyL** [comment]
Thank you @sitatec for the detailed explanation, I'll report it to AOSP instead.

**Comment by TonnyL** [comment]
Issue tracker address: https://issuetracker.google.com/issues/219988993
